Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp

```diff
@@ -523,4 +524,143 @@ namespace UnitTest
+        auto& param = GetParam();
+        EXPECT_THAT(AZ::Quaternion::CreateFromEulerDegreesYXZ(Vector3RadToDeg(param.eular)), IsClose(param.result));
+    }
+
+    INSTANTIATE_TEST_CASE_P(
+        MATH_Quaternion,
+        AngleRadianTestFixtureYXZ,
+        ::testing::Values(
+            EularTestArgs{ AZ::Vector3(Constants::QuarterPi, 0, 0), AZ::Quaternion::CreateRotationY(Constants::QuarterPi) },
+            EularTestArgs{ AZ::Vector3(0, Constants::QuarterPi, 0), AZ::Quaternion::CreateRotationX(Constants::QuarterPi) },
+            EularTestArgs{ AZ::Vector3(0, 0, Constants::QuarterPi), AZ::Quaternion::CreateRotationZ(Constants::QuarterPi) },
+
+            EularTestArgs{ AZ::Vector3(-Constants::QuarterPi, 0, 0), AZ::Quaternion::CreateRotationY(-Constants::QuarterPi) },
+            EularTestArgs{ AZ::Vector3(0, -Constants::QuarterPi, 0), AZ::Quaternion::CreateRotationX(-Constants::QuarterPi) },
+            EularTestArgs{ AZ::Vector3(0, 0, -Constants::QuarterPi), AZ::Quaternion::CreateRotationZ(-Constants::QuarterPi) },
+
+            EularTestArgs{ AZ::Vector3(Constants::QuarterPi, Constants::QuarterPi, 0), AZ::Quaternion::CreateRotationY(Constants::QuarterPi) *
+                AZ::Quaternion::CreateRotationX(Constants::QuarterPi) },
+            EularTestArgs{ AZ::Vector3(0, Constants::QuarterPi, Constants::QuarterPi), AZ::Quaternion::CreateRotationX(Constants::QuarterPi) *
+                AZ::Quaternion::CreateRotationZ(Constants::QuarterPi) },
+            EularTestArgs{ AZ::Vector3(Constants::QuarterPi, 0, Constants::QuarterPi), AZ::Quaternion::CreateRotationY(Constants::QuarterPi) *
+                AZ::Quaternion::CreateRotationZ(Constants::QuarterPi)},
+
+            EularTestArgs{ AZ::Vector3(Constants::HalfPi, 0, Constants::QuarterPi), 
+                AZ::Quaternion::CreateRotationY(Constants::HalfPi) *
+                AZ::Quaternion::CreateRotationZ(Constants::QuarterPi)},
+            EularTestArgs{ AZ::Vector3(-Constants::QuarterPi, -Constants::HalfPi, Constants::QuarterPi), 
+                AZ::Quaternion::CreateRotationY(-Constants::QuarterPi) *
+                AZ::Quaternion::CreateRotationX(-Constants::HalfPi) *
+                AZ::Quaternion::CreateRotationZ(Constants::QuarterPi)},
+            EularTestArgs{ AZ::Vector3(-Constants::QuarterPi, Constants::HalfPi, Constants::TwoOverPi), 
+                AZ::Quaternion::CreateRotationY(-Constants::QuarterPi) *
+                AZ::Quaternion::CreateRotationX(Constants::HalfPi) *
+                AZ::Quaternion::CreateRotationZ(Constants::TwoOverPi)}
+                
+        ));
+
+    using AngleRadianTestFixtureZYX = ::testing::TestWithParam<EularTestArgs>;
+
+    TEST_P(AngleRadianTestFixtureZYX, EularRadiansYXZ) {
+        auto& param = GetParam();
+        EXPECT_THAT(AZ::Quaternion::CreateFromEulerRadiansZYX(param.eular), IsClose(param.result));
+    }
+
+    TEST_P(AngleRadianTestFixtureZYX, EularDegreesYXZ) {
```

Comment:
update: Makes sure test names are updated to `Euler` too (try a find and replace in this file)